R/expect-condition.R

```diff
@@ -582,3 +576,29 @@ cnd_message <- function(x) {
   withr::local_options(rlang_backtrace_on_error = "none")
   conditionMessage(x)
 }
+
+check_condition_dots <- function(
+  regexp = NULL,
+  ...,
+  error_call = caller_env()
+) {
+  if (!is.null(regexp) || missing(...)) {
+    return()
+  }
+
+  dot_names <- ...names()
```

Comment:
ah, in case it matters, I just came across this admonition to continue avoiding `...names()` until 4.1.3 😱 

https://github.com/r-lib/lintr/issues/2502

https://github.com/r-lib/lintr/blob/dc8f14996fc303d40f120eea6d56fb54187dbc9c/R/lint.R#L43